src/Components/Components/src/Auth/PendingAuthenticationState.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Security.Claims;
+
+namespace Microsoft.AspNetCore.Components
+{
+    internal class PendingAuthenticationState : IAuthenticationState
+    {
```

Comment:
I would call this AnonymousAuthenticationState as this represents a user that hasn't been authenticated.

Reply:
TBH I prefer `PendingAuthenticationState` because it doesn't *only* represent a user that hasn't been authenticated: it represents a subset of those, ones who not only haven't been authenticated, but also have an authentication process currently in flight.

In any case this is not part of the public API surface.